### ISSUE TITLE
Incorrect Alignment Returned by Bio.pairwise2.align.localds

### ISSUE THREAD

**Issue by slundberg** [biopython/biopython#279]
I ran across this issue while testing my implementation of Smith-Waterman against BioPython's (version 1.63 on OSX 10.9). The following test case illustrates a non-optimal alignment returned by BioPython. I was following the docs from http://biopython.org/DIST/docs/api/Bio.pairwise2-module.html. I didn't see any open issues documenting this, hence I decided to open this one.

Thanks!

```python
import Bio.pairwise2
from Bio.SubsMat.MatrixInfo import blosum62

print ('K','Q'), 'blosum62 score is',blosum62[('K','Q')]
print ('A','A'), 'blosum62 score is',blosum62[('A','A')]
print ('H','H'), 'blosum62 score is',blosum62[('H','H')]

alignments = Bio.pairwise2.align.localds('VKAHGKKV', 'FQAHCAGV', blosum62, -4, -4)
for a in alignments:
     print Bio.pairwise2.format_alignment(*a)

print "However the score 13 is possible by adding the pairing K-Q..."
```

which for convenience when run produces
```
('K', 'Q') blosum62 score is 1
('A', 'A') blosum62 score is 4
('H', 'H') blosum62 score is 8
VKAHGKKV
  ||
FQAHCAGV
  Score=12

However the score 13 is possible by adding the pairing K-Q...
```

**Comment by peterjc** [comment]
There is currently no active maintainer for ``Bio.pairwise2`` (and there are a few other open issues with it).

**Comment by MarkusPiotrowski** [comment]
The problem seems to be solved with the version I'm acutally working on (not on GitHub yet)
```
('K', 'Q') blosum62 score is 1
('A', 'A') blosum62 score is 4
('H', 'H') blosum62 score is 8
VKAHGKKV
 |||
FQAHCAGV
  Score=13

However the score 13 is possible by adding the pairing K-Q...
```

**Comment by peterjc** [comment]
This will be fixed in the next release (Biopython 1.68) which includes the ``Bio.pairwise2`` work from @MarkusPiotrowski - thank you for the clear test case @slundberg :)
